heartbeat/monitors/wrappers/monitors.go

```diff
@@ -92,3 +97,77 @@ func addMonitorDuration(job jobs.Job) jobs.Job {
 		return cont, err
 	}
 }
+
+// makeAddSummary summarizes the job, adding the `summary` field to the last event emitted.
+func makeAddSummary(id string, numJobs uint16) jobs.JobWrapper {
+	// This is a tricky method. The way this works is that we track the state across jobs in the
+	// state struct here.
+	state := struct {
```

Comment:
As this is all local to `makeAddSummary` I'm good with the implementation as it can't be used anywhere else. Otherwise I would have expected that `state` would have methods to update the fields.